src/api/artifact/controller.go

```diff
@@ -64,6 +65,8 @@ type Controller interface {
 	GetByReference(ctx context.Context, repository, reference string, option *Option) (artifact *Artifact, err error)
 	// Delete the artifact specified by ID. All tags attached to the artifact are deleted as well
 	Delete(ctx context.Context, id int64) (err error)
+	// Copy the artifact whose ID is specified by "from" into the repository
+	Copy(ctx context.Context, repositoryID, from int64) (id int64, err error)
```

Comment:
from(int64) is better than from(*Artifact)?

`from(int64)` will also need to call `get()` internally